### ISSUE TITLE
Add source file limit

### ISSUE THREAD

**Issue by lane55** [pc2ccs/pc2v9#606]
**Is your feature request related to a problem?** 

Yes, team's are allowed to submit HUGE source files

**Feature Description**: 

Add a global source file limit (in K)

Return a CE ??  if a team submits a file that is too large

**Have you considered other ways to accomplish the same thing?** 

**Do you have any specific suggestions for how your feature would be ***implemented*** in PC^2?**

**Additional context**:

**Comment by lane55** [comment]
Analysis: what judgement/response should pc2 give for the limit exceeded?

**Comment by clevengr** [comment]
Regarding the question `what judgement/response should pc2 give`, I would suggest that the limit be enforced on the client (team UI) side -- that is, the UI should not allow a team to proceed with a submission that's "too large".  In that case, there will never be a "response" from PC2 because the PC2 server will never see the submission.

**Comment by clevengr** [comment]
See also Issue #767, which f is either a duplicate of this or which needs a better explanation of what it is getting at that's NOT the same as this issue.

**Comment by johnbrvc** [comment]
> Regarding the question `what judgement/response should pc2 give`, I would suggest that the limit be enforced on the client (team UI) side -- that is, the UI should not allow a team to proceed with a submission that's "too large". In that case, there will never be a "response" from PC2 because the PC2 server will never see the submission.

I agree that the source file limit should be enforced on the submitting client before it even reaches the server, however, it should be noted that there are currently 3 ways to make submissions in PC2.

1. pc2team thick client
2. WTI
3. CLICS API (pc2submit python utility)

Each of these would have to be aware of the limit.  Currently, there is no way to convey this information to the WTI or **pc2submit** utility, so it would have to be queried somehow.  In fact, since **pc2submit** is a python utility, any attempt to enforce the limit there could easily be circumvented by a contestant by copying the python script and editing out the part that limits the source size.  The alternative is to do what we currently do for submission throttling, and that is, reject the submission on the reception of the bad file.  For the WTI, the could be the PC2 API.  For CLICS this could be the event feeder client that is handling the submission.  For pc2team it can just check that browsed-to file for complying with the size.

I am investigating this now (as of 08/13/2025)
